src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -484,27 +486,46 @@ export default class DefaultAudioVideoController
     this.meetingSessionContext.enableSimulcast = this.enableSimulcast;
 
     if (this.enableSimulcast) {
-      const simulcastPolicy = new DefaultSimulcastUplinkPolicy(
-        this.configuration.credentials.attendeeId,
-        this.meetingSessionContext.logger
-      );
-      simulcastPolicy.addObserver(this);
+      let simulcastPolicy = this.meetingSessionContext.videoUplinkBandwidthPolicy;
+      //Make sure the simulcast policy passed in implement the SimulcastUplinkPolicy
+      if (simulcastPolicy && !(simulcastPolicy as SimulcastUplinkPolicy).addObserver) {
+        this.meetingSessionContext.logger.error(
+          'Video uplink bandwidth policy does not implement' +
+            ' SimulcastUplinkPolicy interface. Default to the default simulcast policy.'
+        );
+        simulcastPolicy = undefined;
+      }
+      if (!simulcastPolicy) {
+        simulcastPolicy = new DefaultSimulcastUplinkPolicy(
+          this.configuration.credentials.attendeeId,
+          this.meetingSessionContext.logger
+        );
+      }
+      (simulcastPolicy as SimulcastUplinkPolicy).addObserver(this);
       this.meetingSessionContext.videoUplinkBandwidthPolicy = simulcastPolicy;
 
+      if (!this.meetingSessionContext.videoDownlinkBandwidthPolicy) {
+        this.meetingSessionContext.videoDownlinkBandwidthPolicy = new VideoAdaptiveProbePolicy(
+          this.meetingSessionContext.logger
+        );
+      }
+
       this.meetingSessionContext.videoStreamIndex = new SimulcastVideoStreamIndex(this.logger);
     } else {
       this.meetingSessionContext.enableSimulcast = false;
       this.meetingSessionContext.videoStreamIndex = new DefaultVideoStreamIndex(this.logger);
-      if (!this.meetingSessionContext.videoDownlinkBandwidthPolicy) {
-        this.meetingSessionContext.videoDownlinkBandwidthPolicy = new AllHighestVideoBandwidthPolicy(
+
+      if (!this.meetingSessionContext.videoUplinkBandwidthPolicy) {
+        this.meetingSessionContext.videoUplinkBandwidthPolicy = new NScaleVideoUplinkBandwidthPolicy(
```

Comment:
I guess this means they need to check for browser support before picking the policy?  e.g. on FF if you just set SImulcastUplinkPolicy for everything, it would be completely broken given all the no-op functions that may be called depending on `enableSimulcast`?

Reply:
There are a few reasonable positions to take:

1. Simulcast policies seamlessly degrade to non-simulcast in unsupported browsers. 
2. Simulcast policy constructors throw if they won’t be able to function.
3. 2 plus exposing a way for builders to check (e.g., a static isSupported method on a simulcast policy class).

Reply:
Implement validation (option 1) in DefaultMeetingSession.

Reply:
I guess its fine since at this point almost no builders are writing their own policies.  But technically this implementation requires builders with a custom non-simulcast policy to shim it (or check for support) on their own.